### PR TITLE
Update to support latest level

### LINKED ISSUE #21: Crash bug with latest level
**System Information**
- Aedes: 0.42.5
- NodeJS: 14.12.0
- OS: Ubuntu 20.04 LTS
- Arch: x86_64

**Describe the bug**
`aedes-persistence-level` crashes with latest `level`. 

**To Reproduce**

I've created a repository that reproduces the crash here: https://github.com/ralphtheninja/aedes-crash-with-level
How to reproduce is described in the readme.

**Expected behavior**
It shouldn't crash.

**Additional context**
It only seems to happen when you publish a `qos: 2` message.

### PR DESCRIPTION
**Pull request by ralphtheninja**
Fixes some problems with missing `{ valueEncoding: 'binary' }` in calls to `db.get()`. Without them, we get strings back which makes `msgpack` very unhappy.

Updated tests using latest `level` instead of `levelup` + `memdown`.

Closes https://github.com/moscajs/aedes-persistence-level/issues/21

### CODE REVIEW

**Comment by ralphtheninja** [comment]
Squash when happy.

File: persistence.js
Comment (ralphtheninja) [inline persistence.js:89]: I think this was actually a 'bug'. Technically the `lt` should've been `'retained:\xff'` (with a colon), compared to all other keys, e.g. `SUBSCRIPTIONS` etc.

File: test.js
Comment (ralphtheninja) [inline test.js:10]: `tempy.directory()` is very useful, we use it in `leveldown` and other modules.

Comment (robertsLando) [reply test.js:10]: :+1:

File: persistence.js
Comment (robertsLando) [inline persistence.js]: Could you transform all this including modules require to `const`?

**Review by robertsLando — CHANGES_REQUESTED** [review]

File: persistence.js
Comment (robertsLando) [reply persistence.js:89]: I think it is! :+1:

File: persistence.js
Comment (ralphtheninja) [reply persistence.js]: Sure, no problem. I was thinking about this as well.

File: persistence.js
Comment (ralphtheninja) [inline persistence.js]: `standard` complains about this not being reassigned so `let` -> `const`

File: persistence.js
Comment (ralphtheninja) [inline persistence.js:251]: What's your thoughts on using arrow functions to get rid of `that` variables?

File: persistence.js
Comment (ralphtheninja) [inline persistence.js:273]: I know you can use `const` in for-loops, but I think it looks weird.

File: persistence.js
Comment (ralphtheninja) [inline persistence.js]: We should really use `${INCOMING}${clientId}:${package.messageId}` imo

Comment (robertsLando) [reply persistence.js]: Go for the template syntax yes!

**Review by robertsLando — APPROVED** [review]
lgtm

**Comment by ralphtheninja** [comment]
Not running github actions on PRs?

**Comment by robertsLando** [comment]
> Not running github actions on PRs?

Github bug? Cannot see nothing on actions tab :confused:

**Comment by robertsLando** [comment]
Oh totally missed that :fearful: You have to place github workflow in `.github/workflows/` folder

**Comment by robertsLando** [comment]
@ralphtheninja I have made a pr to fix that #25

File: persistence.js
Comment (robertsLando) [reply persistence.js:251]: @ralphtheninja I prefer to keep `that`, also in all other projects we use this style :)

**Comment by robertsLando** [comment]
Let's wait for #25 and #26 to land then pull master and once ready this can be merged too

**Comment by ralphtheninja** [comment]
> Oh totally missed that fearful You have to place github workflow in `.github/workflows/` folder

Sorry. I knew this, but screwed up anyway :)

**Comment by robertsLando** [comment]
@mcollina last ping request (for today) :)

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by robertsLando** [comment]
@mcollina please add me as a collaborator of this in npm and/or move the package under the moscajs org in order to be able to publish a new major release 

https://www.npmjs.com/package/aedes-persistence-level

**Pull request merged**
